@here/harp-mapview/lib/text/LayoutState.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    DefaultTextStyle,
+    HorizontalAlignment,
+    TextLayoutStyle,
+    VerticalAlignment
+} from "@here/harp-text-canvas";
+
+/**
+ * Version of layout style stored.
+ */
+export enum LayoutUsed {
+    Undefined = 0,
+    ThemeBased,
+    Alternative
```

Comment:
This distinction between ThemeBased and Alternative won't make sense once we have placement alternatives in the style.

Reply:
It also depends from the decision about alignment, if we would like to have alignment working separately, i.e.: text aligned left and placed on the left side of icon. Then it might be useful.
I will see what do to with the states... For now I need at least two (initialized or undefined).

Reply:
I will remove all state based behavior in update and use base / alternative checks on the fly.